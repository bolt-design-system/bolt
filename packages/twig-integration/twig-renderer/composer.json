{
  "name": "bolt-design-system/twig-renderer",
  "type": "library",
  "description": "Twig rendering service used by the Bolt Design System.",
  "version": "2.20.0",
  "license": "MIT",
  "repositories": [
    {
      "type": "path",
	    "url": "../twig-extensions-shared"
	  },
    {
      "type": "path",
      "url": "../twig-extensions-compat"
    }
  ],
  "require": {
    "bolt-design-system/core-php": "*",
    "bolt-design-system/drupal-twig-extensions": "*",
    "twig/twig": "^1.42.5",
    "webmozart/path-util": "*",
    "symfony/finder": "*"
  },
  "scripts": {
    "setup": "@composer install --no-interaction --prefer-dist --no-progress"
  }
}
