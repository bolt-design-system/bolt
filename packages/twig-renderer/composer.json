{
  "name": "bolt-design-system/twig-renderer",
  "type": "library",
  "version": "2.3.0",
  "license": "MIT",
  "require": {
    "bolt-design-system/core-php": "^2.3.0",
    "bolt-design-system/drupal-twig-extensions": "*",
    "twig/twig": "^1.38.2",
    "webmozart/path-util": "^2.3",
    "symfony/finder": "*"
  },
  "scripts": {
    "setup": "@composer install --no-interaction --prefer-dist --no-progress"
  }
}
