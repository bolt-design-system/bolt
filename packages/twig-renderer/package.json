{
  "name": "@bolt/twig-renderer",
  "version": "2.7.0",
  "license": "MIT",
  "description": "Renders Twig templates using an on-the-fly PHP rendering service.",
  "main": "twig-renderer.js",
  "repository": "https://github.com/bolt-design-system/bolt/tree/master/packages/twig-renderer",
  "keywords": [
    "design system",
    "bolt design system",
    "twig",
    "php"
  ],
  "maintainers": [
    {
      "name": "Salem Ghoweri",
      "email": "me@salemghoweri.com",
      "web": "https://github.com/sghoweri"
    }
  ],
  "dependencies": {
    "@basalt/twig-renderer": "^0.12.1",
    "@bolt/build-utils": "^2.7.0",
    "@bolt/drupal-twig-extensions": "^2.7.0",
    "sleep-promise": "^8.0.1"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "setup": "composer run setup"
  },
  "schema": "utils/config.schema.yml",
  "bugs": "https://github.com/bolt-design-system/bolt/issues",
  "homepage": "https://boltdesignsystem.com",
  "gitHead": "3d198feea5f73c63913a956a2a077e229ad8d462"
}
