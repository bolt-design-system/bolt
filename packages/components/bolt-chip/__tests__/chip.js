import { render } from '@bolt/twig-renderer';
const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
const { join } = require('path');
const schema = readYamlFileSync(join(__dirname, '../chip.schema.yml'));
const { url, target, icon, size } = schema.properties;

describe('chip', () => {
  test('basic usage', async () => {
    const results = await render('@bolt-components-chip/chip.twig', {
      text: 'Hello World',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('url usage', async () => {
    const results = await render('@bolt-components-chip/chip.twig', {
      text: 'Has URL',
      url: 'https://pega.com',
      target: '_blank',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('icon usage', async () => {
    const results = await render('@bolt-components-chip/chip.twig', {
      text: 'Icon before',
      icon: {
        name: 'check',
        position: 'before',
      },
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  size.enum.forEach(async (option) => {
    test(`chip size: ${option}`, async () => {
      const results = await render('@bolt-components-chip/chip.twig', {
        text: 'Chip size',
        size: option,
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });
});
