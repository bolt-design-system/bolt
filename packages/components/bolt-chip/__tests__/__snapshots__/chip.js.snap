// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chip basic usage 1`] = `
<bolt-chip>
  <replace-with-grandchildren>
    <span class="c-bolt-chip c-bolt-chip--size-small"
          is="shadow-root"
    >
      <replace-with-children class="c-bolt-chip__text">
        Hello World
      </replace-with-children>
    </span>
  </replace-with-grandchildren>
</bolt-chip>
`;

exports[`chip chip size: medium 1`] = `
<bolt-chip size="medium">
  <replace-with-grandchildren>
    <span class="c-bolt-chip c-bolt-chip--size-medium"
          is="shadow-root"
    >
      <replace-with-children class="c-bolt-chip__text">
        Chip size
      </replace-with-children>
    </span>
  </replace-with-grandchildren>
</bolt-chip>
`;

exports[`chip chip size: small 1`] = `
<bolt-chip size="small">
  <replace-with-grandchildren>
    <span class="c-bolt-chip c-bolt-chip--size-small"
          is="shadow-root"
    >
      <replace-with-children class="c-bolt-chip__text">
        Chip size
      </replace-with-children>
    </span>
  </replace-with-grandchildren>
</bolt-chip>
`;

exports[`chip chip size: xsmall 1`] = `
<bolt-chip size="xsmall">
  <replace-with-grandchildren>
    <span class="c-bolt-chip c-bolt-chip--size-xsmall"
          is="shadow-root"
    >
      <replace-with-children class="c-bolt-chip__text">
        Chip size
      </replace-with-children>
    </span>
  </replace-with-grandchildren>
</bolt-chip>
`;

exports[`chip icon usage 1`] = `
<bolt-chip>
  <replace-with-grandchildren>
    <span class="c-bolt-chip c-bolt-chip--size-small"
          is="shadow-root"
    >
      <replace-with-children class="c-bolt-chip__icon">
        <bolt-icon slot="before"
                   name="check"
        >
          <span class="c-bolt-icon c-bolt-icon--check">
            <svg class="c-bolt-icon__icon"
                 xmlns="http://www.w3.org/2000/svg"
                 viewbox="0 0 24 24"
            >
              <g fill="none"
                 fill-rule="evenodd"
              >
                <path>
                </path>
                <path fill="var(--bolt-icon-primary-color)"
                      d="M19.292 6.29L8.998 16.584 4.703 12.29a1.004 1.004 0 00-1.412 0 1.002 1.002 0 000 1.413l5 5a.995.995 0 001.411 0l11-11a.99.99 0 000-1.413 1.001 1.001 0 00-1.41 0z"
                >
                </path>
              </g>
            </svg>
          </span>
        </bolt-icon>
      </replace-with-children>
      <replace-with-children class="c-bolt-chip__text">
        Icon before
      </replace-with-children>
    </span>
  </replace-with-grandchildren>
</bolt-chip>
`;

exports[`chip url usage 1`] = `
<bolt-chip url="https://pega.com"
           target="_blank"
>
  <replace-with-grandchildren>
    <a href="https://pega.com"
       target="_blank"
       class="c-bolt-chip c-bolt-chip--link c-bolt-chip--size-small"
       is="shadow-root"
    >
      <replace-with-children class="c-bolt-chip__text">
        Has URL
      </replace-with-children>
    </a>
  </replace-with-grandchildren>
</bolt-chip>
`;
